python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/sensor_builder.py

```diff
@@ -0,0 +1,264 @@
+            f"***********"
+        )
+        try:
+            cursor = (
+                deserialize_value(context.cursor, DbtCloudPollingSensorCursor)
+                if context.cursor
+                else DbtCloudPollingSensorCursor()
+            )
+        except Exception as e:
+            context.log.info(f"Failed to interpret cursor. Starting from scratch. Error: {e}")
+            cursor = DbtCloudPollingSensorCursor()
+        current_date = get_current_datetime()
+        current_offset = cursor.offset or 0
+        finished_at_start = (
+            cursor.finished_at_start
+            or (current_date - timedelta(seconds=START_LOOKBACK_SECONDS)).timestamp()
+        )
+        finished_at_end = cursor.finished_at_end or current_date.timestamp()
+        sensor_iter = materializations_from_batch_iter(
+            context=context,
+            finished_at_start=finished_at_start,
+            finished_at_end=finished_at_end,
+            offset=current_offset,
+            workspace=workspace,
+            dagster_dbt_translator=dagster_dbt_translator,
+        )
+
+        all_asset_events: list[AssetMaterialization] = []
+        all_check_keys: set[AssetCheckKey] = set()
+        latest_offset = current_offset
+        repository_def = check.not_none(context.repository_def)
+        while get_current_datetime() - current_date < timedelta(seconds=MAIN_LOOP_TIMEOUT_SECONDS):
+            batch_result = next(sensor_iter, None)
+            if batch_result is None:
+                context.log.info("Received no batch result. Breaking.")
+                break
+            all_asset_events.extend(batch_result.asset_events)
+
+            all_check_keys.update(
+                check_keys_for_asset_keys(repository_def, batch_result.all_asset_keys_materialized)
+            )
+            latest_offset = batch_result.idx
+
+        if batch_result is not None:  # pyright: ignore[reportPossiblyUnboundVariable]
+            new_cursor = DbtCloudPollingSensorCursor(
+                finished_at_start=finished_at_start,
+                finished_at_end=finished_at_end,
+                offset=latest_offset + 1,
+            )
+        else:
+            # We have completed iteration for this range
+            new_cursor = DbtCloudPollingSensorCursor(
+                finished_at_start=finished_at_end,
+                finished_at_end=None,
+                offset=0,
+            )
+
+        context.update_cursor(serialize_value(new_cursor))
+
+        context.log.info(
+            f"************"
+            f"Exiting sensor for dbt Cloud workspace for account {workspace.account_name}, "
+            f"project {workspace.project_name} and environment {workspace.environment_name}"
+            f"***********"
+        )
+        return SensorResult(
+            asset_events=sorted_asset_events(all_asset_events, repository_def),
+            run_requests=[RunRequest(asset_check_keys=list(all_check_keys))]
```

Comment:
Requesting runs for asset check keys shouldn't be necessary anymore; the whole reason we did that is because airlift landed before DA for checks. I think you can just rip this bit out.